### PR TITLE
[JAVA] Use JUnit 4 features to log Java tests log

### PR DESCRIPTION
**Pull request by dkurt**
Example output:
```
testGetBlobFromFloat(BlobTests) - OK
testGetBlob(BlobTests) - OK
testInputName(CNNNetworkTests) - OK
testAddOutput(CNNNetworkTests) - OK
testReshape(CNNNetworkTests) - OK
testReadNetworkIncorrectXmlPath(IECoreTests) - OK
testInitIECore(IECoreTests) - OK
testLoadNetworDeviceConfig(IECoreTests) - OK
testReadNetworkIncorrectBinPath(IECoreTests) - OK
testReadNetwork(IECoreTests) - OK
testLoadNetwork(IECoreTests) - OK
testLoadNetworkWrongDevice(IECoreTests) - OK
testReadNetworkXmlOnly(IECoreTests) - OK
testSetCompletionCallback(InferRequestTests) - OK
testGetPerformanceCounts(InferRequestTests) - OK
testStartAsync(InferRequestTests) - OK
testSetLayout(InputInfoTests) - OK
testSetPrecision(InputInfoTests) - OK
```

### CODE REVIEW

File: inference-engine/ie_bridges/java/tests/InferRequestTests.java [hunk 1]
```diff
@@ -12,15 +16,16 @@
     InferRequest inferRequest;
     boolean completionCallback;
 
-    @Override
-    protected void setUp() {
+    @Before
+    public void setUp() {
         core = new IECore();
         net = core.ReadNetwork(modelXml);
         executableNetwork = core.LoadNetwork(net, "CPU");
```
Comment (likholat) [inline inference-engine/ie_bridges/java/tests/InferRequestTests.java:23]: I think, we shouldn't create ```core```, ```net``` and ```executableNetwork``` before each test, we can make it once.

File: inference-engine/ie_bridges/java/tests/IECoreTests.java [hunk 2]
```diff
@@ -66,9 +71,11 @@ public void testLoadNetworDeviceConfig(){
         assertTrue(executableNetwork instanceof ExecutableNetwork);
     }
 
+    @Test
     public void testLoadNetworkWrongDevice(){
```
Comment (likholat) [inline inference-engine/ie_bridges/java/tests/IECoreTests.java:75]: Missing space before brace (for all tests in this file).

File: inference-engine/ie_bridges/java/tests/IECoreTests.java [hunk 3]
```diff
@@ -1,57 +1,62 @@
...
     
-    @Override
-    protected void setUp() {
-        core = new IECore();
-        exceptionMessage = "";
-    }
-
+    @Test
     public void testInitIECore(){
         assertTrue(core instanceof IECore);
```
Comment (likholat) [inline inference-engine/ie_bridges/java/tests/IECoreTests.java:14]: We don't need this test, ```core``` will always be instance of ```IECore```.

File: inference-engine/ie_bridges/java/tests/BlobTests.java [hunk 4]
```diff
@@ -13,6 +15,7 @@ public void testGetBlob(){
         Assert.assertArrayEquals(blob.getTensorDesc().getDims(), dimsArr);
     }
 
+    @Test
     public void testGetBlobFromFloat(){
```
Comment (likholat) [inline inference-engine/ie_bridges/java/tests/BlobTests.java:19]: Missing space before bracket (for both tests).

Comment (ilyachur) [reply inference-engine/ie_bridges/java/tests/BlobTests.java:19]: I think that the code style checker can solve such issues :)

File: inference-engine/ie_bridges/java/tests/InferRequestTests.java (hunk 1 above)
Comment (ilyachur) [reply inference-engine/ie_bridges/java/tests/InferRequestTests.java:23]: Do you have a plan to extend these tests for other plugins?

**Review by ilyachur — APPROVED** [review]
In general LGTM. One question: Do we run these tests in the pre-commit or in other validation scope?

File: inference-engine/ie_bridges/java/tests/InferRequestTests.java (hunk 1 above)
Comment (dkurt) [reply inference-engine/ie_bridges/java/tests/InferRequestTests.java:23]: Added a parameter

File: inference-engine/ie_bridges/java/tests/IECoreTests.java (hunk 2 above)
Comment (dkurt) [reply inference-engine/ie_bridges/java/tests/IECoreTests.java:75]: Fixed

File: inference-engine/ie_bridges/java/tests/IECoreTests.java (hunk 3 above)
Comment (dkurt) [reply inference-engine/ie_bridges/java/tests/IECoreTests.java]: Removed

**Review by likholat — APPROVED** [review]

**Pull request merged**